src/Dumplie/Infrastructure/InMemory/Customer/InMemoryProducts.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Infrastructure\InMemory\Customer;
+
+use Dumplie\Domain\Customer\Exception\ProductNotFoundException;
+use Dumplie\Domain\Customer\Product;
+use Dumplie\Domain\Customer\Products;
+
+final class InMemoryProducts implements Products
+{
+    /**
+     * @var array
+     */
+    private $products;
+
+    /**
+     * @param array|Product[] $products
+     */
+    public function __construct(array $products = [])
+    {
+        $this->products = [];
+
+        foreach ($products as $product) {
+            if (!$product instanceof Product) {
+                throw new \InvalidArgumentException();
+            }
+
+            $this->products[(string) $product->sku()] = $product;
+        }
+    }
+
+    /**
+     * @param string $sku
+     * @return Product
+     * @throws ProductNotFoundException
+     */
+    public function getBySku(string $sku) : Product
+    {
+        if (!array_key_exists((string) $sku, $this->products)) {
```

Comment:
is there a need to convert `$sku` to `string` while there is a `string` typehint?